### PR TITLE
fine-tune IconButtons

### PR DESCRIPTION
**Pull request by michelengelen**
Fine tuning the IconButton and adding a `compact` prop to adhere to the new compact version of the IconButton

### CODE REVIEW

File: src/components/icon-button/IconButton.root.tsx [hunk 1]
```diff
@@ -3,29 +3,38 @@ import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components
...
+        destructive,
+        disabled,
+        theme: { palette, action, text, animation },
     }: ThemedStyledProps<PIconButtonRoot, TTheme>): FlattenSimpleInterpolation => {
         const isDefault = !inverted && !destructive && !toggled;
         const { main, contrast } = destructive && !toggled ? palette.alert : palette.primary;
 
+        // this solution is not ideal and should probably be refactored in the
+        // `applyPadding` and `applyMargin` mixin-function
+        let spacingTokenIndex = SPACING_TOKENS.indexOf(ICON_BUTTON_DEFINITIONS[size].spacing);
```
Comment (nevyangelova) [inline src/components/icon-button/IconButton.root.tsx:32]: nit: I am not sure its a good idea to use this expensive operation and it takes some time to figure out whats going on. Isn't it better to have a separate spacing prop for the compact versions in the definitions map?

**Review by nevyangelova — COMMENTED** [review]
Otherwise LGTM!

File: src/components/icon-button/IconButton.root.tsx (hunk 1 above)
Comment (michelengelen) [reply src/components/icon-button/IconButton.root.tsx:32]: probably ... as mentioned in the comment this is not ideal and should be rethought

Comment (michelengelen) [reply src/components/icon-button/IconButton.root.tsx:32]: any ideas?

Comment (nevyangelova) [reply src/components/icon-button/IconButton.root.tsx:32]: I was trying to find a way to mutate the array with one of the sorting methods but I think the simplest solution might be something like: 
 
 ```
const ICON_BUTTON_DEFINITIONS: { [size in TIconButtonSizeToken]: TIconButtonDefinition } = {
    xs: {
       spacingCompact: 0,
        spacing: 75,
        iconSize: 12,
        fontSize: 75,
    },
    ...
  }
 ```
 
 If I understand the requirement correctly? We don't need to move the index and we can just take the value directly

Comment (michelengelen) [reply src/components/icon-button/IconButton.root.tsx]: @nevyangelova I did apply your approach for now, but once we get to have more components that have a `compact` version (candidates can potentially be `Button`, `TextField` or `Checkbox` etc.) we should rethink the probability of adding it to the layout mixins (`applyMargin` and `applyPadding`)

**Review by nevyangelova — APPROVED** [review]

**Pull request merged**